cli/command/cli.go

```diff
@@ -407,9 +418,55 @@ type ClientInfo struct {
 	DefaultVersion  string
 }
 
-// NewDockerCli returns a DockerCli instance with IO output and error streams set by in, out and err.
-func NewDockerCli(in io.ReadCloser, out, err io.Writer, isTrusted bool, containerizedFn func(string) (clitypes.ContainerizedClient, error)) *DockerCli {
-	return &DockerCli{in: NewInStream(in), out: NewOutStream(out), err: err, contentTrust: isTrusted, newContainerizeClient: containerizedFn}
+// NewDockerCli returns a DockerCli instance with all operators applied on it.
+func NewDockerCli(ops ...DockerCliOperator) *DockerCli {
+	cli := &DockerCli{}
+	cli.Apply(ops...)
+	return cli
+}
+
+// WithStandardStreams sets a cli in, out and err streams with the standard streams.
+func WithStandardStreams() DockerCliOperator {
+	return func(cli *DockerCli) {
+		cli.in = NewInStream(os.Stdin)
+		cli.out = NewOutStream(os.Stdout)
+		cli.err = os.Stderr
+	}
+}
+
+// WithInputStream sets a cli input stream.
+func WithInputStream(in io.ReadCloser) DockerCliOperator {
+	return func(cli *DockerCli) {
+		cli.in = NewInStream(in)
+	}
+}
+
+// WithOutputStream sets a cli output stream.
+func WithOutputStream(out io.Writer) DockerCliOperator {
+	return func(cli *DockerCli) {
+		cli.out = NewOutStream(out)
+	}
+}
+
+// WithErrorStream sets a cli error stream.
+func WithErrorStream(err io.Writer) DockerCliOperator {
+	return func(cli *DockerCli) {
+		cli.err = err
+	}
+}
+
+// WithContentTrust enables content trust on a cli.
+func WithContentTrust() DockerCliOperator {
```

Comment:
FWIW I think the default should be to obey the `DOCKER_CONTENT_TRUST` variable (encapsulated via `contentTrustEnabled()` today) and there should be one (or two) of these to force it on or off explicitly.

Reply:
👍 for default contentTrust value initialized using the env variable, plus to enable/disable explicitly, I would have the signature be `func WithContentTrust(enabled bool) DockerCliOperator`

Reply:
Fixed 👍